server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -216,7 +225,13 @@
      */
     public static final Setting<Boolean> COMPRESS_SETTING = Setting.boolSetting("compress", true, Setting.Property.NodeScope);
 
-    private final Settings settings;
+    /**
+     * When set to {@code true}, {@link #bestEffortConsistency} will be set to {@code true} and concurrent modifications of the repository
+     * contents will not result in the repository being marked as corrupted.
+     * Note: This setting is intended as a backwards compatibility solution for 7.x and will go away in 8.
+     */
+    public static final Setting<Boolean> ALLOW_CONCURRENT_MODIFICATION =
+        Setting.boolSetting("allow_concurrent_modifications", false, Setting.Property.Deprecated);
```

Comment:
How do we deal with this?

Reply:
Good point, thank you. This relates to the `bestEffortConsistency` https://github.com/crate/crate/blob/d3106e9fdf33156d2e0c3175c17adc129e913e56/server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java#L280
and would AFAIK only cover the case where multiple clusters write into the same repository all other cases are handled here 
https://github.com/crate/crate/blob/d3106e9fdf33156d2e0c3175c17adc129e913e56/server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java#L363
Do we have the case where multiple clusters write into the same repository ? Otherwise we could remove the setting.

Reply:
> Do we have the case where multiple clusters write into the same repository ?

Afaik we don't mention this as a supported feature. So I guess we could remove it.


@proddata @WalBeh do you see use-cases for this?

Reply:
Confused how this would even really be possible? Same repository meaning also same root-folder?
We might transition repositories when moving from one cluster to another, but I am not aware of parallel use.

Not sure if relevant to **Cloud** (@SStorm). Afaik we share S3 repos within projects, but I would guess in separate folders.

Reply:
Clusters do share s3 buckets, however each cluster's "repository" lives in own subfolder.

Reply:
Yes, sharing the repository would mean sharing the same bucket (folder).

Reply:
Thanks for the feedback.

Reply:
> Yes, sharing the repository would mean sharing the same bucket (folder).

Sorry, i meant sharing the same folder.